spacy_llm/backends/integration/hf/base.py

```diff
@@ -0,0 +1,105 @@
+import abc
+import warnings
+from typing import Iterable, TypeVar, Any, Dict
+from ....compat import has_torch, has_transformers, has_accelerate, torch
+from thinc.compat import has_torch_cuda_gpu
+
+# Type of prompts returned from Task.generate_prompts().
+_PromptType = TypeVar("_PromptType")
+# Type of responses returned from query function.
+_ResponseType = TypeVar("_ResponseType")
+
+
+class HuggingFaceBackend(abc.ABC):
```

Comment:
Does this really need to be an abstract class? It feels to me like the implemented methods could also be in something like `hf_util`

Reply:
I worry that we're introducing a lot of abstract layers here...

Reply:
Ok, let's keep as is :-)